services/filemanager/miniomanager.go

```diff
@@ -79,7 +82,12 @@ func (manager *MinioManager) GetObjectNameFromLocation(location string) (string,
 
 //TODO complete this
 func (manager *MinioManager) GetDownloadKeyFromFileLocation(location string) string {
-	return location
+	parsedUrl, err := url.Parse(location)
+	if err != nil {
+		fmt.Println("error while parsing location url: ", err)
+	}
+	trimedUrl := strings.TrimLeft(parsedUrl.Path, "/")
+	return strings.TrimPrefix(trimedUrl, fmt.Sprintf(`%s/`, manager.Config.Bucket))
```

Comment:
What if s3 path style is not used ? Is it always used for minio ? Was it working before ?

Reply:
minio by always uses pathStyle url, as per my knowledge. and previously we didn't had a working code for it.